### PR TITLE
qt6: do not define default_options for modules

### PR DESCRIPTION
**Pull request by ericLemanissier**
closes conan-io/conan-center-index#20565
fixes conan-io/conan-center-index#20383

Specify library name and version:  **qt/6.***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
@matheusgomes28 this should hopefully be a more general solution than https://github.com/conan-io/conan-center-index/pull/20565. Can you please try it and confirm ?

**Comment by matheusgomes28** [comment]
@ericLemanissier I'll try a clean build after work today. I see you haven't yet changed the module dependency lists on the top of the file. Maybe I can push that later on.

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6530988792)

**Comment by ericLemanissier** [comment]
Do you mean [the addition of `ShaderTools` to `qtmultimedia` module dependencies](https://github.com/conan-io/conan-center-index/pull/20565/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R1204) ?
I could not find in qtmultimedia repo which target links to shadertools. Maybe you can show us what makes you think it does or the build failure which happens without this change ?

**Review by jwillikers — APPROVED** [review]

**Comment by matheusgomes28** [comment]
> Do you mean [the addition of `ShaderTools` to `qtmultimedia` module dependencies](https://github.com/conan-io/conan-center-index/pull/20565/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R1204) ? I could not find in qtmultimedia repo which target links to shadertools. Maybe you can show us what makes you think it does or the build failure which happens without this change ?

I assumed it was necessary as the dependency is declared in `qtmodules6.5.2.conf`:

```toml
[submodule "qtmultimedia"]
	depends = qtbase qtshadertools
	recommends = qtdeclarative qtquick3d
	path = qtmultimedia
	url = ../qtmultimedia.git
	branch = 6.5.2
	status = addon
```

And as described in #20383 , you cannot build `qtmultimedia` without `qtshadertools`.

**Comment by ericLemanissier** [comment]
Not always, it can be used only at compile time

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6530988792)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/6530988792)

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -214,12 +214,21 @@ def configure(self):
         def _enablemodule(mod):
             if mod != "qtbase":
                 setattr(self.options, mod, True)
-            for req in self._get_module_tree[mod]["depends"]:
-                _enablemodule(req)
+                for req in self._get_module_tree[mod]["depends"]:
+                    _enablemodule(req)
 
+        # enable all modules which are
+        # - required by a module explicitely enabled by the consumer
```
Comment (matheusgomes28) [inline recipes/qt/6.x.x/conanfile.py:221]:
```suggestion
        # - required by a module explicitly enabled by the consumer
```

File: recipes/qt/6.x.x/conanfile.py [hunk 2]
```diff
@@ -214,12 +214,21 @@ def configure(self):
...
         for module in self._get_module_tree:
-            if self.options.get_safe(module):
+            if getattr(self.options, module):
                 _enablemodule(module)
+        
+        # disable all modules which are:
+        # - not explicitely enabled by the consumer and
```
Comment (matheusgomes28) [inline recipes/qt/6.x.x/conanfile.py:227]:
```suggestion
        # - not explicitly enabled by the consumer and
```

**Comment by matheusgomes28** [comment]
> Not always, it can be used only at compile time

@ericLemanissier  Yes, you can have build dependency & also runtime dependencies... The lines in question (calling the `_create_module(...)` functions) on the recipe seems to be representing modules created for _build_ dependencies, hence the suggestion.

Furthermore, I have tried the new recipe just now (replacing the cached one with the recipe in this PR). Seems to work fine on `6.5.2`, will try the other `6.5.x` tomorrow. Will report back if anything is broken, otherwise assume it's all workingfine.

**Comment by ericLemanissier** [comment]
@matheusgomes28 OK, so I understand the misunderstanding: the _create_module helper is defining information related to the consumption of qt modules. That's the purpose of package_info(self). So it only describes run-time dependencies for consumers of qt. Build requirements are not relevant here.

**Review by MartinDelille — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**